src/Cryptol/Eval/Concrete.hs

```diff
@@ -65,45 +63,47 @@ toExpr prims t0 v0 = findOne (go t0 v0)
 
 
   go :: AST.Type -> Value -> ChoiceT Eval Expr
-  go ty val = case (tNoUser ty, val) of
-    (TRec tfs, VRecord vfs) -> do
-      -- NB, vfs first argument to keep their display order
-      res <- zipRecordsM (\_lbl v t -> go t =<< lift v) vfs tfs
-      case res of
-        Left _ -> mzero -- different fields
-        Right efs -> pure (ERec efs)
-    (TCon (TC (TCTuple tl)) ts, VTuple tvs) -> do
-      guard (tl == (length tvs))
-      ETuple `fmap` (zipWithM go ts =<< lift (sequence tvs))
-    (TCon (TC TCBit) [], VBit True ) -> return (prim "True")
-    (TCon (TC TCBit) [], VBit False) -> return (prim "False")
-    (TCon (TC TCInteger) [], VInteger i) ->
-      return $ ETApp (ETApp (prim "number") (tNum i)) ty
-    (TCon (TC TCIntMod) [_n], VInteger i) ->
-      return $ ETApp (ETApp (prim "number") (tNum i)) ty
-    (TCon (TC TCRational) [], VRational (SRational n d)) ->
-      do let n' = ETApp (ETApp (prim "number") (tNum n)) (TCon (TC TCInteger) [])
-             d' = ETApp (ETApp (prim "number") (tNum d)) (TCon (TC TCInteger) [])
-         return $ EApp (EApp (prim "ratio") n') d'
-    (TCon (TC TCFloat) [eT,pT], VFloat i) ->
-      pure (floatToExpr prims eT pT (bfValue i))
-
-    (TCon (TC TCSeq) [a,b], VSeq 0 _) -> do
```

Comment:
Weird that this special case is here... it seems unnecessary, but do we know why it is here?

Reply:
It looked to me like that special case was completely redundant; I'm not sure why it was there.

Reply:
Looks like that code has been bouncing around with that special case since 156771a5d5fe405d7e033928a1ac25d43ebf3642.  Dosen't seem like there's a specific reason. *shrug*